WordPress/src/main/java/org/wordpress/android/ui/accounts/login/LoginEmailFragment.java

```diff
@@ -0,0 +1,397 @@
+        Pattern emailRegExPattern = Patterns.EMAIL_ADDRESS;
+        Matcher matcher = emailRegExPattern.matcher(email);
+
+        return matcher.find() && email.length() <= MAX_EMAIL_LENGTH;
+    }
+
+    private final OnClickListener mNextClickListener = new OnClickListener() {
+        @Override
+        public void onClick(View v) {
+            next(getEmail());
+        }
+    };
+
+    @Override
+    public void afterTextChanged(Editable s) {
+    }
+
+    @Override
+    public void beforeTextChanged(CharSequence s, int start, int count, int after) {
+    }
+
+    @Override
+    public void onTextChanged(CharSequence s, int start, int before, int count) {
+        UpdateNextButton();
+        mEmailEditTextLayout.setError(null);
+    }
+
+    private void UpdateNextButton() {
+        if (fieldsFilled()) {
+            mNextButton.setEnabled(true);
+        } else {
+            mNextButton.setEnabled(false);
+        }
+    }
+
+    private boolean fieldsFilled() {
+        return EditTextUtils.getText(mEmailEditText).trim().length() > 0;
+    }
+
+    protected boolean isUserDataValid() {
+        final String email = EditTextUtils.getText(mEmailEditText).trim();
+        boolean retValue = true;
+
+        if (TextUtils.isEmpty(email)) {
+            showEmailError(R.string.required_field);
+            retValue = false;
+        }
+
+        return retValue;
+    }
+
+    private void showEmailError(int messageId) {
+        mEmailEditTextLayout.setError(getString(messageId));
+        mEmailEditText.post(new Runnable() {
+            @Override
+            public void run() {
+                mEmailEditText.requestFocus();
+            }
+        });
+    }
+
+    @Override
+    public void onStart() {
+        super.onStart();
+        mDispatcher.register(this);
+    }
+
+    @Override
+    public void onStop() {
+        super.onStop();
+        mDispatcher.unregister(this);
+    }
+
+    private void showEmailCheckProgressDialog() {
+        startProgress(getActivity().getString(R.string.checking_email));
+    }
+
+    @Override
+    protected void startProgress(String message) {
+        mNextButton.setEnabled(false);
+        mProgressDialog = ProgressDialog.show(getActivity(), "", message, true, true,
+                new DialogInterface.OnCancelListener() {
+            @Override
+            public void onCancel(DialogInterface dialog) {
+                UpdateNextButton();
+            }
+        });
+    }
+
+    @Override
+    protected void endProgress() {
+        if (mProgressDialog != null) {
+            mProgressDialog.cancel();
+        }
+
+        // nullify the reference to denote there is no operation in progress
+        mProgressDialog = null;
+
+        UpdateNextButton();
+    }
+
+    // OnChanged events
+
+    @SuppressWarnings("unused")
+    @Subscribe(threadMode = ThreadMode.MAIN)
+    public void onAvailabilityChecked(OnAvailabilityChecked event) {
+        endProgress();
```

Comment:
> there is no check which one you've just received the callback for
That's not quite accurate. Notice how `handleEmailAvailabilityEvent()` is picking up the verified email via its method parameter, not some global/instance variable. Effectively, the `OnAvailabilityChecked` event _does_ hold and propagate the checked email address.

Having said that, truth is that the cancelable dialog doesn't offer much anyway. Took that from the SignInFragment by the way. I could demote it to a non-cancelable one but even as it is, it causes no error and the user can actually try another email address while "waiting". I'm sure this is an edge case so, let me know if you'd prefer to demote it and I'll do it. Thanks!